h2o-algos/src/main/java/hex/glm/GLMMetricBuilder.java

```diff
@@ -38,33 +41,36 @@
   final boolean _computeMetrics;
   public GLMMetricBuilder(String[] domain, double [] ymu, GLMWeightsFun glmf, int rank, boolean computeMetrics, boolean intercept){
     super(domain == null?0:domain.length, domain);
-    _rank = rank;
     _glmf = glmf;
+    _rank = rank;
     _computeMetrics = computeMetrics;
     _intercept = intercept;
     _ymu = ymu;
     if(_computeMetrics) {
-      switch(_glmf._family){
-        case binomial:
-        case quasibinomial:
-          _metricBuilder = new MetricBuilderBinomial(domain);
-          break;
-        case multinomial:
-          _metricBuilder = new MetricBuilderMultinomial(domain.length,domain);
-          ((MetricBuilderMultinomial)_metricBuilder)._priorDistribution = ymu;
-          break;
-        case ordinal:
-          _metricBuilder = new MetricBuilderOrdinal(domain.length,domain);
-          ((MetricBuilderOrdinal)_metricBuilder)._priorDistribution = ymu;
-          break;
-        default:
-          _metricBuilder = new MetricBuilderRegression();
-          break;
+      if (domain!=null && domain.length==1 && domain[0].contains("HGLM")) {
```

Comment:
@wendycwong is there a better way to detect that we are running HGLM?

Reply:
Michalk:  I don't have access to parms and I don't want to break it for other GLM....